src/iinit.cpp

```diff
@@ -43,27 +44,37 @@ const char *AUTH_OPENID_SCHEME = "openid";
  */
 int
 mkrodsdir() {
-    char dirName[NAME_LEN];
-    int mode;
-    char *getVar;
-#ifdef windows_platform
-    getVar = iRODSNt_gethome();
-#else
-    getVar = getenv( "HOME" );
-#endif
-    rstrcpy( dirName, getVar, NAME_LEN );
-    rstrcat( dirName, "/.irods", NAME_LEN );
-    mode = 0700;
-#ifdef _WIN32
-    iRODSNt_mkdir( dirName, mode );
-#else
-    int error_code = mkdir( dirName, mode );
-    int errsv = errno;
-    if ( error_code != 0 && errsv != EEXIST ) {
-        rodsLog( LOG_NOTICE, "mkdir failed in mkrodsdir with error code %d", error_code );
+    namespace fs = std::filesystem;
+
+    const char* home_dir{std::getenv("HOME")};
+    if (home_dir == nullptr) {
+        fmt::print(stderr, "environment variable HOME not set!\n");
```

Comment:
Find myself hesitating on the exclamation point.

In general, I don't think we've been too excited in the logs and in error messages to the user.